metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -1166,23 +1168,45 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> Source:
         config = KafkaConnectSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    def get_connector_payload(self) -> Dict:
+        connector_response = self.session.get(
+            f"{self.config.connect_uri}/connectors",
+        )
+        try:
+            connector_response.raise_for_status()
+            payload = connector_response.json()
+            return payload
+        except requests.exceptions.HTTPError as e:
+            try:
+                payload = connector_response.json()
+                logger.error(f"Error message: {payload}")
+            except json.JSONDecodeError:
+                logger.error(f"Error message: {connector_response.text}")
+            logger.error(f"Error {e} during fetching connectors. Skipping...")
+            raise e
+
     def get_connectors_manifest(self) -> List[ConnectorManifest]:
         """Get Kafka Connect connectors manifest using REST API.
         Enrich with lineages metadata.
         """
         connectors_manifest = list()
 
-        connector_response = self.session.get(
-            f"{self.config.connect_uri}/connectors",
-        )
-
-        payload = connector_response.json()
+        payload = self.get_connector_payload()
 
         for c in payload:
             connector_url = f"{self.config.connect_uri}/connectors/{c}"
             connector_response = self.session.get(connector_url)
             manifest = connector_response.json()
-            connector_manifest = ConnectorManifest(**manifest)
+            try:
+                connector_manifest = ConnectorManifest(**manifest)
+            except TypeError as e:
+                logger.error(
```

Comment:
use self.report.warning(..., exc=e)

that will automatically log the stack trace when debug mode is enabled

We have some internal docs on this here https://www.notion.so/acryldata/Error-reporting-in-ingestion-5343cc6ea0c84633b38070d1a409c569?pvs=4